### PR TITLE
[feat]: [CDS-51650]: fixing mongo index

### PR DESCRIPTION
**Pull request by RajBaviskar**
## Describe your changes

https://user-images.githubusercontent.com/7086440/215703403-1773a1d6-967e-4323-88a0-935ee098cf27.mov

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43257)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/moduleversioninfo/runnable/UpdateVersionInfoTask.java [hunk 1]
```diff
@@ -93,9 +96,22 @@ private void checkVersionChange() {
...
+                              .and(ModuleVersionInfoKeys.microservicesVersionInfo)
+                              .is(module.getMicroservicesVersionInfo())
+                              .and(ModuleVersionInfoKeys.displayName)
+                              .is(module.getDisplayName())
+                              .and(ModuleVersionInfoKeys.lastModifiedAt)
+                              .is(module.getLastModifiedAt())
+                              .and(ModuleVersionInfoKeys.releaseNotesLink)
+                              .is(module.getReleaseNotesLink());
+      Update update = update(ModuleVersionInfoKeys.version, module.getVersion());
+      mongoTemplate.upsert(new Query(criteria), update, ModuleVersionInfo.class);
```
Comment (abhinav-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/moduleversioninfo/runnable/UpdateVersionInfoTask.java:113]: upsert should have on create as well

Comment (RajBaviskar) [reply 120-ng-manager/src/main/java/io/harness/ng/moduleversioninfo/runnable/UpdateVersionInfoTask.java]: I actually changed the update with the whole Object. Tested out Upsert is creating the collection as well. refer recording.

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti0

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43257)* status: 2 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43257)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43257#-:-NN8Qo6M6A-N6sDFjmVP:bnfp4nl)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti1

**Comment by RajBaviskar** [comment]
trigger ti3

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti0

**Comment by RajBaviskar** [comment]
trigger ti0

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43257)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43257)* status: :shipit: complete! ✓

**Pull request merged**